pkg/controllers/management/auth/globalroles/globalrole_handler.go

```diff
@@ -196,6 +217,110 @@ func (gr *globalRoleLifecycle) reconcileCatalogRole(globalRole *v3.GlobalRole) e
 	return nil
 }
 
+// reconcileNamespacedRoles ensures that Roles exist in each namespace of NamespacedRules
+func (gr *globalRoleLifecycle) reconcileNamespacedRoles(globalRole *v3.GlobalRole) error {
+	var returnError error
+
+	// For collecting all the roles that should exist for the GlobalRole
+	roleUIDs := []types.UID{}
+
+	for ns, rules := range globalRole.NamespacedRules {
+		_, err := gr.nsLister.Get("", ns)
+		if err != nil {
+			returnError = multierror.Append(returnError, errors.Wrapf(err, "couldn't get namespace %s", ns))
+			continue
+		}
+		// Check if the role exists
+		roleName := globalRole.Name + "-" + ns
+		role, err := gr.rLister.Get(ns, roleName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+				continue
+			}
+			newRole := &v1.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      roleName,
+					Namespace: ns,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: globalRole.APIVersion,
+							Kind:       globalRole.Kind,
+							Name:       globalRole.Name,
+							UID:        globalRole.UID,
+						},
+					},
+					Labels: map[string]string{
+						grOwnerLabel: wrangler.SafeConcatName(globalRole.Name),
+					},
+				},
+				Rules: rules,
+			}
+			createdRole, err := gr.rClient.Create(newRole)
+			if err == nil {
+				roleUIDs = append(roleUIDs, createdRole.UID)
+				continue
+			} else if !apierrors.IsAlreadyExists(err) {
+				returnError = multierror.Append(returnError, err)
+				continue
+			} else {
+				// In the case that the role already exists, we fall through to check that the rules are correct
+				role = newRole
```

Comment:
Do we want to set role equal to the new role here? If the role already exists, the version of the role which exists in k8s could have different rules than the one constructed in memory here, and in this case we would be checking the integrity of the one constructed in memory.